pages/categorization.vue

```diff
@@ -0,0 +1,521 @@
+				},								
+				
+				// Local reference to the page names in the store
+				pageNames: this.$store.getters.pageNames,
+
+				paintClasses: {
+
+					paint0: "column-paint-0",
+					paint1: "column-paint-1",
+					paint2: "column-paint-2",
+					paint3: "column-paint-3",
+					paint4: "column-paint-4",
+					paintDefault: "column-paint-default"
+				},
+
+				// Whether or not page has enabled access to the annotation page
+				readyForNextStepFlag: false,
+
+				// Data for the coloring listgroup
+				recommendedColumns: {
+					
+					names: [
+
+						"Subject ID",
+						"Age",
+						"Sex",
+						"Diagnosis",
+						"Assessment Tool"
+					],
+					backgroundColors: [
+
+						"rgb(164,208,90)",
+						"rgb(127,23,167)",
+						"rgb(70,76,174)",
+						"rgb(236,197,50)",
+						"rgb(128,1,1)"
+					],
+					foregroundColors: [
+
+						"black",
+						"white",
+						"white",
+						"black",
+						"white"
+					]
+				}
+			}
+		},
+
+		computed: {	
+
+			tableDataFromTsvAndOrJson() {
+
+				// 0. Check that there is tsv and json data in the data store
+				let tsvFile = this.$store.state.pageData.home.tsvFile;
+				let jsonFile = this.$store.state.pageData.home.jsonFile;
+				if ( null == tsvFile && null == jsonFile )
+					return [];
+
+				console.log("Point 1");
+				console.log("tsvFile: " + tsvFile);
+				console.log("jsonFile: " + jsonFile);
+
+				// Uses both tsv and json data
+				if ( null != jsonFile ) {
+
+					console.log("Point 2a");
+
+					// 1. Produce an array of dicts
+					var tsvJsonDictArray = [];
+
+					// A. Each dict has a header entry from the tsv file
+					let headerFields = [];
+					let tsvJsonIndex = 0;
+					let tsvJsonIndexMap = {};
+					for ( let headerField in tsvFile[0] ) {
+
+						// I. Save the header field in a list
+						headerFields.push(headerField);
+
+						// II. Save an index map for quick location of column and description
+						tsvJsonIndexMap.headerField = tsvJsonIndex;
+						tsvJsonIndex += 1;
+
+						// III. Save a new dict for this column and description
+						tsvJsonDictArray.push({
+
+							"column": headerField,
+							"description": "",
+							"primary-key": tsvJsonIndex - 1
+							//"bColor": this.$store.state.columnCategorization.default.bColor,
+							//"fColor": this.$store.state.columnCategorization.default.fColor,
+						});
+					}
+
+					console.log("Point 2b");
+
+					// B. and a corresponding "description" column that is (possibly) sourced from the json file
+					for ( let json_column in jsonFile ) {
+
+						// I. Save a lowercase version of the current json key
+						let jsonColumnLowercase = json_column.toLowerCase();
+
+						// II. Try to match the json key with one in the tsv file
+						if ( headerFields.includes(jsonColumnLowercase) ) {
+
+							for ( let index = 0; index < tsvJsonDictArray.length; index++ ) {
+
+								// NOTE: Advanced column name matching here between tsv and json? J. Armoza 01/26/22
```

Comment:
Do you mean case-sensitive matching here?

Reply:
This will actually a spot where a call to your intelligent matching from Python code can sit. Since the tsv and json keys may not match or may be disparate.